src/dmd/link.d

```diff
@@ -251,7 +251,11 @@ public int runLINK()
             }
 
             VSOptions vsopt;
-            vsopt.initialize();
+            // if a runtime library (msvcrtNNN.lib) from the mingw folder is selected explicitly, do not detect VS and use lld
+            if (global.params.mscrtlib.length <= 6 ||
+                global.params.mscrtlib[0..6] != "msvcrt" || !isdigit(global.params.mscrtlib[6]))
```

Comment:
This look like it will match msvcrt869675436, not just msvcrt100

Reply:
Yeah, I wanted to cover the versions that we might distribute once we upgrade to the mingw folders to what's build by https://github.com/dlang/installer/pull/346. LDC already has these, i.e. msvcrt80/90/90d/100/110/120/120d/120_app.

Reply:
Should I defer that to when these are actually used?

Reply:
Na I think it's more than fine to already check for it as it doesn't hurt at all and we will go this way.

Reply:
> I wanted to cover the versions that we might distribute

You may want to include a `vcruntime` prefix too then; `vcruntime140` is the UCRT/VC14 equivalent (unless you're planning to rename it).

Reply:
> You may want to include a `vcruntime` prefix too then; `vcruntime140` is the UCRT/VC14 equivalent (unless you're planning to rename it).

Not sure, we'll have to revisit this when we get to that. I'm also not happy with msvcrt.lib in the LDC folder,, as this is in conflict with the regular MS library.